### PR TITLE
Update XGBoost to 1.7.0

### PR DESCRIPTION
**Pull request by hcho3**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by hcho3** [comment]
@conda-forge-admin, please rerender

**Comment by hcho3** [comment]
@jakirkham Can we merge this?

**Comment by valeriupredoi** [comment]
hey guys, sorry to drop in here totally unannounced, but I'm not sure you guys are aware that various tests for building docs on Readthedocs are failing because `xgboost` version mismatch between conda-forge and pypi, so it'd be nice if you guys merged this one the sooner the better. Cheers muchly :beer:

File: recipe/0001-Force-endian-flag-in-cross-compilation-mode.patch [hunk 1]
```diff
@@ -1,14 +1,5 @@
-From f8ec7ff592140327a7c822718b0edf027d0c6b3d Mon Sep 17 00:00:00 2001
-From: "Uwe L. Korn" <uwe.korn@quantco.com>
-Date: Sun, 24 Oct 2021 20:11:23 +0200
```
Comment (xhochy) [inline recipe/0001-Force-endian-flag-in-cross-compilation-mode.patch:3]: Please keep the patch headers

Comment (hcho3) [reply recipe/0001-Force-endian-flag-in-cross-compilation-mode.patch]: Done

**Review by xhochy — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**